pkg/reconcilermanager/controllers/reposync_controller.go

```diff
@@ -934,16 +934,18 @@ func (r *RepoSyncReconciler) validateNamespaceSecret(ctx context.Context, repoSy
 	return validateSecretData(authType, secret)
 }
 
-func (r *RepoSyncReconciler) upsertRoleBinding(ctx context.Context, reconcilerRef, rsRef types.NamespacedName) (client.ObjectKey, error) {
+func (r *RepoSyncReconciler) upsertRoleBinding(ctx context.Context, reconcilerRef, rsRef types.NamespacedName, labelMap map[string]string) (client.ObjectKey, error) {
 	rbRef := client.ObjectKey{
 		Namespace: rsRef.Namespace,
 		Name:      RepoSyncPermissionsName(),
 	}
 	childRB := &rbacv1.RoleBinding{}
 	childRB.Name = rbRef.Name
 	childRB.Namespace = rbRef.Namespace
+	delete(labelMap, metadata.SyncNameLabel)
```

Comment:
This modifies the map that is being passed in, which will have a side effect on the subsequent lines of code that use this map.